API/trade_trees/controllers/trade_trees_controller.py

```diff
@@ -63,20 +75,28 @@ def put_trade_tree():
                 return flask.jsonify(result), 400
 
             tree = self.parser.parse_args()
-            result = self.service.put_trade_tree(tree)
+            result = self.tree_service.put_trade_tree(tree)
             return flask.jsonify(result), 204
 
         @blueprint.route('/api/trade_tree/<id>', methods=['DELETE'])
+        @auth.login_required
         def delete_trade_tree(id):
-            # TODO: Introduce authorization.
-
-            result = self.service.delete_trade_tree(id)
+            result = self.tree_service.delete_trade_tree(id)
             return flask.jsonify(result=result), 200
 
         @blueprint.route('/api/trade_tree/evaluate/<id>', methods=['GET'])
+        @auth.login_required
         def evaluate_trade_tree(id):
             # TODO: Introduce authorization.
-            result = self.service.evaluate_trade_tree(id)
+            result = self.tree_service.evaluate_trade_tree(id)
             return flask.jsonify(result=result), 200
 
+        @blueprint.route('/api/create-user', methods=['POST'])
```

Comment:
This has to go to a separated controller

Reply:
Done